### PR TITLE
[AV-15933] updated examples for public API

### PR DESCRIPTION
**Pull request by nidhi07kumar**
I have updated sample Golang use cases for Couchbase Public API.  

The sample cases have been tested. Test results and observations are updated in Jira ticket.

### CODE REVIEW

**Review by Rhymond — CHANGES_REQUESTED** [review]
Please use `go fmt` to format files

File: go/examples/createuser/main.go [hunk 1]
```diff
@@ -26,18 +27,29 @@ func main() {
...
+		log.Fatal(err)
+	}
+
+	_ = utils.PrettyPrint(resp)
+
+	allBucket := UserCreatePayload{
+		AllBucketsAccess: "data_reader",
+	}
+
+	resp, err = c.Do(http.MethodPost, "/v2/clusters/"+os.Args[1]+"/users/"+os.Args[2], allBucket)
```
Comment (Rhymond) [inline go/examples/createuser/main.go]: We're changing it to `put`
```suggestion
	resp, err = c.Do(http.MethodPut, "/v2/clusters/"+os.Args[1]+"/users/"+os.Args[2], allBucket)
```

Comment (nidhi07kumar) [reply go/examples/createuser/main.go]: @Rhymond Updated it to Put. But, in the docs it is still a Post request.

**Review by Rhymond — APPROVED** [review]
🚢  it!

**Pull request merged**